src/QueryHandler.cc

```diff
@@ -368,13 +370,32 @@ void QueryHandler::process_query(protobufs::queryMessage& proto_query,
 
         Json::Value& tx_responses = pmgd_query.run();
 
-        if (tx_responses.size() != root.size()) { // error
+        if (!tx_responses.isArray()) {
+            Json::StyledWriter writer;
+            std::cerr << "PMGD Response:" << std::endl;
+            std::cerr << writer.write(tx_responses) << std::endl;
```

Comment:
Is this debug trace output that shouldn't be committed?

Reply:
We have these as logging all over the query handling. We have pending a real logging mechanism.